### PR TITLE
refactor: seperate score result from rating report for cutechess output

### PR DESCRIPTION
**Pull request by gahtan-syarif**
mimic cutechess behavior and also makes it more easily parseable i think

### CODE REVIEW

File: src/matchmaking/output/output_cutechess.hpp [hunk 1]
```diff
@@ -35,6 +35,16 @@ class Cutechess : public IOutput {
...
 
+        std::cout << ss.str() << std::flush;
+    }
+
+    void printElo(const Stats& stats, const std::string& first,
+                  const std::string& second) override {
+        const elo::EloWDL elo(stats);
+
+         //get rid of unused parameter compiler warning
+        if (false) std::cout << first << second;
```
Comment (Disservin) [inline src/matchmaking/output/output_cutechess.hpp:46]: Remove the name from the parameter list

File: src/matchmaking/output/output_fastchess.hpp [hunk 2]
```diff
@@ -19,6 +19,13 @@ class Fastchess : public IOutput {
         std::cout << "--------------------------------------------------\n";
     };
 
+    void printResult(const Stats& stats, const std::string& first,
+                     const std::string& second) override {
+        //do nothing
+        //get rid of unused parameter compiler warning
+        if (false) std::cout << first << second << stats.wins;
```
Comment (Disservin) [inline src/matchmaking/output/output_fastchess.hpp:26]: Same

**Review by Disservin — CHANGES_REQUESTED** [review]

File: src/matchmaking/output/output_cutechess.hpp (hunk 1 above)
Comment (gahtan-syarif) [reply src/matchmaking/output/output_cutechess.hpp:46]: wouldnt it just error?

Comment (Disservin) [reply src/matchmaking/output/output_cutechess.hpp:46]:
```
(const Stats&, const std::string&, const std::string&)
```

Comment (gahtan-syarif) [reply src/matchmaking/output/output_cutechess.hpp]: updated

File: src/matchmaking/output/output_fastchess.hpp (hunk 2 above)
Comment (gahtan-syarif) [reply src/matchmaking/output/output_fastchess.hpp]: updated

**Pull request merged**